app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.health = health
+        self.hidden = False
+        self.name = name
+
+        if self.health > 0:
+            Animal.alive.append(self)
+
+    def __repr__(self) -> str:
+        animals_repr = (f"{{Name: {self.name}, "
+                        f"Health: {self.health}, Hidden: {self.hidden}}}")
+        return animals_repr
+
+
+class Herbivore(Animal):
+    def hide(self) -> None:
+        if not self.hidden:
+            self.hidden = True
+        else:
+            self.hidden = False
+
+
+class Carnivore(Animal):
+    def bite(self, herbivore: str) -> None:
```

Comment:
The `bite` method's parameter `herbivore` is typed as `str`, but it should be an instance of `Herbivore`. Change the type annotation from `str` to `Herbivore`.